aiida/orm/users.py

```diff
@@ -84,6 +83,15 @@ def normalize_email(email: str) -> str:
             email = '@'.join([email_name, domain_part.lower()])
         return email
 
+    @property
+    def is_default(self) -> bool:
+        """Return whether the user is the default user.
+
+        :returns: Boolean, ``True`` if the user is the default, ``False`` otherwise.
+        """
+        default_user = self.collection.get_default()
```

Comment:
So this corresponds to the default user of the profile? I.e. the user with email:

```
get_manager().get_profile().default_user_email
```

Reply:
Ah, I guess the message of commit three answers my question. ^^

In my experience, one profile will rarely have multiple users. I don't think profiles are shared commonly like that. But I suppose when you import data from another user, you'll have multiple users stored. I do wonder if users ever switch default user. ^^

Reply:
Yes. If you look at `UserCollection.get_default` it does
```python
return self.backend.default_user
```
and the `StorageBackend` implementation implements `default_user` as 
```python

    @property
    def default_user(self) -> Optional['User']:
        """Return the default user for the profile, if it has been created.

        This is cached, since it is a frequently used operation, for creating other entities.
        """
        from aiida.orm import QueryBuilder, User

        if self._default_user is None and self.profile.default_user_email:
            query = QueryBuilder(self).append(User, filters={'email': self.profile.default_user_email})
            self._default_user = query.first(flat=True)
        return self._default_user
```

Reply:
So there you can see it queries the user based on the value of `profile.default_user_email`.

Reply:
Hmm, so every time you create/store a node, you have to set the default user, but this requires a query?

Reply:
Yes, but that is why it is cached on the backend, so once it is loaded, you only do the query once

Reply:
Oh, right. My bad 🙈

![image](https://github.com/aiidateam/aiida-core/assets/25607960/c70264a7-ba2c-4538-8538-6d15e75485d7)